### PR TITLE
Fix allocation of VMs with multiple clusters

### PR DESCRIPTION
**Pull request by BryanMLima**
### Description

This PR fixes a scenario where if the first cluster verified by ACS does not have any compatible hosts, an exception is thrown, ignoring the verification of possible compatible hosts in other clusters.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

- Cluster 1: Host A
- Cluster 2: Host B

Using the environment above, I added the flexible host tag `false` to Host A, and tried to deploy a VM; which would always result on an incompatible host for Cluster 1. Prior to this patch, the `CloudRuntimeException` is thrown with the message `No suitable host found for vm [...].` and Host B is not verified.

After applying this patch, the message `No suitable host found for VM [%s] in %s.` is still logged, however, Host B is verified for compatibility and the VM is deployed as expected.

#### How did you try to break this feature and the system with this change?

### CODE REVIEW

**Comment by weizhouapache** [comment]
@BryanMLima 
target to main or 4.18/4.19 ?

**Comment by BryanMLima** [comment]
> @BryanMLima target to main or 4.18/4.19 ?

Thanks, I rebased to `4.19` as the problem arose from PR #7489.

**Comment by BryanMLima** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocator.java [hunk 1]
```diff
@@ -132,9 +131,9 @@ public List<Host> allocateTo(VirtualMachineProfile vmProfile, DeploymentPlan pla
             return new ArrayList<Host>();
         }
 
-        if (s_logger.isDebugEnabled()) {
-            s_logger.debug("Looking for hosts in dc: " + dcId + "  pod:" + podId + "  cluster:" + clusterId);
-        }
+        String paramAsStringToLog = String.format("zone [%s], pod [%s], cluster [%s]", dcId, podId, clusterId);
+
+        s_logger.debug(String.format("Looking for hosts in %s.", paramAsStringToLog));
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocator.java:136]: I see no reason to remove this guard.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by rohityadavcloud — APPROVED** [review]
LGTM - but this need some (manual) testing

**Comment by BryanMLima** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by JoaoJandre — APPROVED** [review]
CLGTM, didn't test it

**Comment by BryanMLima** [comment]
@blueorangutan package

**Review by borisstoyanov — APPROVED** [review]
LGTM, manually checked the changes.

**Pull request merged**